logger/logger.go

```diff
@@ -33,30 +47,61 @@ func NewLogger(config LoggerConfig) (hclog.Logger, error) {
 	}), nil
 }
 
-func getLogFileWriter(logFilePath string, appendFile bool) (*os.File, error) {
-	logFilePath = strings.Trim(logFilePath, " ")
-	if logFilePath == "" {
-		return nil, nil
+func newRotatingLogger(config LoggerConfig) (hclog.Logger, error) {
+	logFilePathTrimmed, _, err := createLogDir(config.LogFilePath)
+	if err != nil {
+		return nil, err
 	}
 
-	logFileDirectory := filepath.Dir(logFilePath)
+	lumber := &lumberjack.Logger{
+		Filename:   logFilePathTrimmed,
+		MaxSize:    config.RotatingLogerConfig.MaxSizeInMB,
+		MaxBackups: config.RotatingLogerConfig.MaxBackups,
+		MaxAge:     config.RotatingLogerConfig.MaxAgeInDays,
+		Compress:   config.RotatingLogerConfig.Compress,
+	}
 
-	if err := common.CreateDirSafe(logFileDirectory, 0770); err != nil {
+	return hclog.New(&hclog.LoggerOptions{
+		Name:       config.Name,
+		Level:      config.LogLevel,
+		Output:     lumber,
+		JSONFormat: config.JSONLogFormat,
+	}), nil
+}
+
+func getLogFileWriter(logFilePath string, appendFile bool) (*os.File, error) {
+	logFilePathTrimmed, logFileDirectory, err := createLogDir(logFilePath)
+	if err != nil {
 		return nil, err
 	}
 
 	if !appendFile {
 		suffix := strings.Replace(strings.Replace(time.Now().UTC().Format(time.RFC3339), ":", "_", -1), "-", "_", -1)
-		logFileName := filepath.Base(logFilePath)
+		logFileName := filepath.Base(logFilePathTrimmed)
 
 		if parts := strings.SplitN(logFileName, ".", 2); len(parts) == 1 {
 			logFileName = fmt.Sprintf("%s_%s", parts[0], suffix)
 		} else {
 			logFileName = fmt.Sprintf("%s_%s.%s", parts[0], suffix, parts[1])
 		}
 
-		logFilePath = filepath.Join(logFileDirectory, logFileName)
+		logFilePathTrimmed = filepath.Join(logFileDirectory, logFileName)
+	}
+
+	return os.OpenFile(logFilePathTrimmed, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0660)
+}
+
+func createLogDir(logFilePath string) (string, string, error) {
+	logFilePathTrimmed := strings.Trim(logFilePath, " ")
```

Comment:
I would not consider empty path as error (i think it is same as current directory). Also I would not return logFilePathTrimmed from this function. Why do we trim space character exactly?

Reply:
Will keep previous behaviour